layers/igmp.go

```diff
@@ -16,18 +16,210 @@ import (
 
 type IGMPType uint8
 
+const (
+	MembershipQuery    IGMPType = 0x11 // General or group specific query
+	MembershipReportV1 IGMPType = 0x12 // Version 1 Membership Report
+	MembershipReportV2 IGMPType = 0x16 // Version 2 Membership Report
+	LeaveGroup         IGMPType = 0x17 // Leave Group
+	MembershipReportV3 IGMPType = 0x22 // Version 3 Membership Report
+)
+
+// String conversions for IGMP message types
+func (i IGMPType) String() string {
+	switch i {
+	case MembershipQuery:
+		return "IGMP Membership Query"
+	case MembershipReportV1:
+		return "IGMPv1 Membership Report"
+	case MembershipReportV2:
+		return "IGMPv2 Membership Report"
+	case MembershipReportV3:
+		return "IGMPv3 Membership Report"
+	case LeaveGroup:
+		return "Leave Group"
+	default:
+		return ""
+	}
+}
+
+type IGMPQuery interface {
+	decodeResponse(data []byte) error
+}
+
 // IGMP is the packet structure for IGMP messages.
 type IGMP struct {
 	BaseLayer
-	Type            IGMPType
```

Comment:
This change to the IGMP type breaks backwards compatibility:  If someone's already using type IGMP and references, for example, the RobustnessValue field, this change will break their code.  We try very hard to not break existing code with new commits.

Is the current IGMP struct version-specific?  If so, maybe just create a new struct IGMPv2 or whatever and leave IGMP around so old code continues to run.  Thoughts?